src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -836,39 +1180,46 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
-  public freezeMinting = async (params: TxParams) => {
+  public freezeIssuance = async (params: FreezeIssuanceParams) => {
     assert.assert(
-      await (await this.featureRegistryContract()).getFeatureStatus.callAsync(Features.FreezeMintingAllowed),
+      await (await this.featureRegistryContract()).getFeatureStatus.callAsync(Feature.FreezeMintingAllowed),
       'FreezeMintingAllowed Feature Status not enabled',
     );
-    assert.assert(!(await this.mintingFrozen()), 'Minting already frozen');
+    assert.assert(await this.isIssuable(), 'Issuance frozen');
     assert.assert(
       functionsUtils.checksumAddressComparision(
         await this.owner(),
         (await this.web3Wrapper.getAvailableAddressesAsync())[0],
       ),
       'Msg sender must be owner',
     );
-    return (await this.contract).freezeMinting.sendTransactionAsync(params.txData, params.safetyFactor);
+    return (await this.contract).freezeIssuance.sendTransactionAsync(
+      params.signature,
+      params.txData,
+      params.safetyFactor,
+    );
   };
 
-  public mint = async (params: MintParams) => {
+  public issue = async (params: IssueParams) => {
     assert.isNonZeroETHAddressHex('investor', params.investor);
     await this.checkOnlyOwner(params.txData);
-    assert.assert(!(await this.mintingFrozen()), 'Minting is frozen');
-    return (await this.contract).mint.sendTransactionAsync(
+    assert.assert(await this.isIssuable(), 'Issuance frozen');
+    return (await this.contract).issue.sendTransactionAsync(
       params.investor,
       valueToWei(params.value, await this.decimals()),
+      params.data,
       params.txData,
       params.safetyFactor,
     );
   };
 
-  public mintWithData = async (params: MintWithDataParams) => {
+  public issueByPartition = async (params: IssueByPartitionParams) => {
     assert.isNonZeroETHAddressHex('investor', params.investor);
     await this.checkOnlyOwner(params.txData);
-    assert.assert(!(await this.mintingFrozen()), 'Minting is frozen');
-    return (await this.contract).mintWithData.sendTransactionAsync(
+    assert.assert(await this.isIssuable(), 'Issuance frozen');
+    assert.assert(params.partition === 'UNLOCKED', 'Invalid Partition');
```

Comment:
Same as above, partitions should be part of an enum

Also, this check should probably be done on a separate function `isValidPartition` that compares the value to all possible partition values. This will make this easier to extend if/when more partitions are added